app/controllers/posts_controller.rb

```diff
@@ -30,6 +24,13 @@ def create
     end
   end
 
+  def index
+    @user = User.find(params[:user_id])
+    @posts = Post.where(author_id: @user.id).includes(:comments, :user).paginate(page: params[:page], per_page: 10)
+  end
+
```

Comment:
- [optional ] 👋 Hey there! Thanks for the PR, and great work on the code! 🎉

- [ ] In your `post#index` action, you've used `.includes(:comments, :user)` to address the N+1 query problem, which is a great approach.


####  However, Using this suggestion:
```
def index
    @user = User.find(params[:user_id])
    @posts = @user.posts.includes(:comments) 
end
```
 - [ ] We can make the code more concise and efficient by directly accessing the user's posts through `@user.posts`. This not only simplifies the code but also results in more efficient database queries.

- [ ] So, if you're open to a suggestion, I'd recommend going with the second approach as it's more readable and can potentially improve performance. It also leverages ActiveRecord associations, which is a best practice when working with Rails.

Keep up the good work! 🚀